contentcuration/contentcuration/static/js/edit_channel/channel_list/tests/channelListStore.spec.js

```diff
@@ -1,177 +0,0 @@
-import _ from 'underscore';
-import store from './../vuex/store';
-import { ListTypes } from './../constants';
-import { Invitations } from './data';
-
-const Backbone = require('backbone');
-
-/*
-  TODO: there are some issues trying to mock jquery.ajax as it
-  throws a `TypeError: Cannot read property 'prototype' of undefined`
-  due to how it interacts with Backbone. Added TODOs where we'll need
-  to test
-*/
-
-describe('channelListStore', () => {
-  let channel;
-  beforeEach(() => {
-    channel = { id: 'test', name: 'channel', main_tree: 'node' };
-    store.commit('channel_list/RESET_STATE');
-  });
-  afterEach(() => {
-    jest.clearAllMocks();
-  });
-
-  describe('channel actions', () => {
-    describe('loadChannelList', () => {
-      it('should get list of channels', () => {
-        _.each(_.values(ListTypes), type => {
-          store.dispatch('channel_list/loadChannelList', type);
-          // TODO: make sure ChannelListUrls[type] endpoint is called
-        });
-      });
-      describe('adding channels to the store', () => {
-        let channels;
-        let payload1;
-        let payload2;
-
-        beforeEach(() => {
-          store.commit('channel_list/RESET_STATE');
-          channels = store.state.channel_list.channels;
-          payload1 = {
-            listType: ListTypes.EDITABLE,
-            channels: [{ id: 'channel 1' }, { id: 'channel 2' }],
-          };
-          payload2 = {
-            listType: ListTypes.VIEW_ONLY,
-            channels: [{ id: 'channel 1' }, { id: 'channel 3' }],
-          };
-        });
-
-        it('should merge the channels to one list', () => {
-          store.commit('channel_list/SET_CHANNEL_LIST', payload1);
-          expect(channels).toHaveLength(2);
-          store.commit('channel_list/SET_CHANNEL_LIST', payload2);
-          expect(channels).toHaveLength(3);
-        });
-
-        it('should mark relevant list types as true', () => {
-          store.commit('channel_list/SET_CHANNEL_LIST', payload1);
-          store.commit('channel_list/SET_CHANNEL_LIST', payload2);
-          expect(_.find(channels, { [ListTypes.EDITABLE]: true, id: 'channel 1' })).toBeTruthy();
-          expect(_.find(channels, { [ListTypes.EDITABLE]: true, id: 'channel 2' })).toBeTruthy();
-          expect(_.find(channels, { [ListTypes.EDITABLE]: false, id: 'channel 3' })).toBeTruthy();
-          expect(_.find(channels, { [ListTypes.VIEW_ONLY]: true, id: 'channel 1' })).toBeTruthy();
-          expect(_.find(channels, { [ListTypes.VIEW_ONLY]: false, id: 'channel 2' })).toBeTruthy();
-          expect(_.find(channels, { [ListTypes.VIEW_ONLY]: true, id: 'channel 3' })).toBeTruthy();
-        });
-      });
-    });
-
-    describe('star actions', () => {
-      it('addStar should post a star request', () => {
```

Comment:
I have a bunch of unfinished tests here because mocking ajax caused issues with backbone. As you go through this, could you include the tests listed here?